ingestors/kafka/src/main/scala/hydra/kafka/util/ConsumerGroupsOffsetConsumer.scala

```diff
@@ -0,0 +1,223 @@
+                                                                  consumerOffsetsOffsetsTopicConfig: Subject,
+                                                                  kafkaInternalTopic: String,
+                                                                  dvsConsumersTopic: Subject,
+                                                                  bootstrapServers: String,
+                                                                  commonConsumerGroup: String
+                                                                ): F[Unit] = {
+    val dvsConsumerOffsetStream = kafkaClientAlgebra.consumeMessagesWithOffsetInfo(consumerOffsetsOffsetsTopicConfig.value, uniquePerNodeConsumerGroup, commitOffsets = false)
+
+    for {
+      schemaRegistryClient <- schemaRegistryAlgebra.getSchemaRegistryClient
+      deferred <- Deferred[F, PartitionOffsetMap]
+      hydraConsumerOffsetsOffsetsLatestOffsets <- kafkaAdminAlgebra.getLatestOffsets(consumerOffsetsOffsetsTopicConfig.value).map(_.map(l => l._1.partition -> l._2.value))
+      hydraConsumerOffsetsOffsetsCache <- Ref[F].of[PartitionOffsetMap](hydraConsumerOffsetsOffsetsLatestOffsets.filter(_._2 == 0))
+      initialPartitionCache <- initializePartitions(kafkaAdminAlgebra, kafkaInternalTopic)
+      backgroundProcess <- Concurrent[F].start(getOffsetsToSeekTo(initialPartitionCache, deferred, dvsConsumerOffsetStream,hydraConsumerOffsetsOffsetsLatestOffsets, hydraConsumerOffsetsOffsetsCache))
+      partitionMap <- deferred.get
+      _ <- backgroundProcess.cancel
+      _ <- Concurrent[F].start(consumerOffsetsToInternalOffsets(kafkaInternalTopic, dvsConsumersTopic.value, bootstrapServers, commonConsumerGroup, schemaRegistryClient, partitionMap, consumerOffsetsOffsetsTopicConfig.value))
+    } yield ()
+  }
+
+  // Gets partition info for the kafkaInternalTopic "__consumer_offsets" and initializes each to offset to 0
+  private def initializePartitions[F[_]: ConcurrentEffect](kafkaAdminAlgebra: KafkaAdminAlgebra[F], kafkaInternalTopic: String): F[Ref[F, PartitionOffsetMap]] = {
+    for {
+      consumerOffsetsLatestOffsets <- kafkaAdminAlgebra.getLatestOffsets(kafkaInternalTopic).map { m =>
+        m.map(p => (p._1.partition, 0L))
+      }
+      cache <- Ref[F].of(consumerOffsetsLatestOffsets)
+    } yield cache
+  }
+
+  private def getConsumerGroupDeserializer[F[_]: Sync, A](byteBufferToA: ByteBuffer => A): Deserializer[F, Option[A]] =
+    Deserializer.delegate[F, Option[A]] {
+      (_: String, data: Array[Byte]) => {
+        Try(byteBufferToA(Bytes(data).toByteBuffer)).toOption
+      }
+    }.suspend
+
+  private def getSerializer[F[_]: Sync, A](schemaRegistryClient: SchemaRegistryClient)(isKey: Boolean): Serializer[F, A] =
+    Serializer.delegate[F, A] {
+      val serializer = {
+        val se = new KafkaAvroSerializer(schemaRegistryClient)
+        se.configure(Map(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG -> "").asJava, isKey)
```

Comment:
I believe you answered this before, but just to double check, we don't need a url here?

Reply:
No. It's required to have something, but it ignores the value anyway, so empty string it is.